### PR TITLE
[ISSUE #3515] Do some code optimization[AuthTokenUtils]

### LINKED ISSUE #3515: [Enhancement] Do some code optimization[AuthTokenUtils]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

located at:
eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java
analysis and explanation:
a. The logic of "authTokenByPublicKey" and "helloTaskAuthTokenByPublicKey" method in lines "54-87" and "99-132" is the same, so it can be extracted as one method.
b. Variable 'validationKeyBytes' initializer 'new byte[0]' is redundant.[line 66,111]
c. Argument 'publicKeyUrl' might be null.[line 68,113]
image

d. 'if' statement can be simplified.[line 146]
image

e. The "{}" is redundant.[line 90,134]
image

### Describe the solution you'd like

a. Repeat the logic code extracted as a method.
b. Remove redundant initializers.
c. Use 'Objects.requireNonNull()' to avoid an NPE.
image

d. Replace this if-then-else statement by a single return statement
image

e. Remove the extra "{}".
image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by kyooosukedn**
Fixes #3515 .

### Motivation

*Code Optimization on AuthTokenUtils.java*
*Remove redundancy*

### Modifications

*Remove duplicate code and redundancies*

### Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by Alonexc** [comment]
image
Please check the checkstyle.

**Comment by kyooosukedn** [comment]
> image Please check the checkstyle.

Hi, how do I get those warnings? I ran ./gradlew check and did code inspection on file AuthTokenUtils.java and found nothing,  I have imported checkstyle.xml though

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/HTTPClientHandler.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ void list(HttpExchange httpExchange) throws IOException {
             });
 
             String result = JsonUtils.toJSONString(getClientResponseList);
-            httpExchange.sendResponseHeaders(200, result.getBytes().length);
+            httpExchange.sendResponseHeaders(200, result.getBytes(Constants.DEFAULT_CHARSET).length);
```
Comment (Alonexc) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/HTTPClientHandler.java:142]: Here `Constants.DEFAULT_CHARSET` needs to be imported:
org.apache.eventmesh.common.Constants;

Comment (kyooosukedn) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/HTTPClientHandler.java:143]: Thanks :)

**Comment by kyooosukedn** [comment]
Hi @Alonexc Is there something else that needs to be done?

**Comment by Pil0tXia** [comment]
Please resolve conflicts.

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java [hunk 2]
```diff
@@ -138,14 +74,44 @@ public static boolean authAccess(AclProperties aclProperties) {
         String topic = aclProperties.getTopic();
 
         Object topics = aclProperties.getExtendedField("topics");
-
+      
         if (!(topics instanceof Set)) {
```
Comment (Pil0tXia) [inline eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java:78]: Please remove these redundant whitespaces here.

**Review by Pil0tXia — COMMENTED** [review]
I didn't see these changes required in #3515:

>b. Remove redundant initializers.
>c. Use 'Objects.requireNonNull()' to avoid an NPE.
>d. Replace this if-then-else statement by a single return statement
>e. Remove the extra "{}".

Please supplement.😊

Besides, please fix checkstyle warnings: https://github.com/apache/eventmesh/actions/runs/7481706701/job/20372145878

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java (hunk 2 above)
Comment (kyooosukedn) [reply eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java:78]: Thanks for your review, will do :)

Comment (Pil0tXia) [reply eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java]: Is this PR ready for review now?😊

Comment (kyooosukedn) [reply eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java]: Yes, if I am not mistaken :)

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java [hunk 3]
```diff
@@ -138,14 +75,58 @@ public static boolean authAccess(AclProperties aclProperties) {
         String topic = aclProperties.getTopic();
 
         Object topics = aclProperties.getExtendedField("topics");
-
         if (!(topics instanceof Set)) {
```
Comment (Pil0tXia) [inline eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java:78]: I think this line is better not removed to maintain the formatting.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

**Comment by kyooosukedn** [comment]
Hi @Pil0tXia, I have corrected as you suggested. Thank u for the review :)

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java [hunk 4]
```diff
@@ -146,6 +81,50 @@ public static boolean authAccess(AclProperties aclProperties) {
         Set<String> groupTopics = TypeUtils.castSet(topics, String.class);
 
         return groupTopics.contains(topic);
+
+    }
```
Comment (Pil0tXia) [inline eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java:85]: We generally won't leave a blank line at the end of a method.

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java [hunk 5]
```diff
@@ -146,6 +81,50 @@ public static boolean authAccess(AclProperties aclProperties) {
...
+            throw new AclException("no such RSA algorithm!", e);
+        } catch (InvalidKeySpecException e) {
+            throw new AclException("invalid public key spec!", e);
+        } catch (JwtException e) {
+            throw new AclException("invalid token!", e);
+        }
+
     }
 
 }
```
Comment (Pil0tXia) [inline eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java:130]: Redundant lines here too.

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

File: eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java (hunk 4 above)
Comment (kyooosukedn) [reply eventmesh-security-plugin/eventmesh-security-auth-token/src/main/java/org/apache/eventmesh/auth/token/impl/auth/AuthTokenUtils.java]: gotcha

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

**Review by Pil0tXia — DISMISSED** [review]
LGTM

**Review by Pil0tXia — APPROVED** [review]

**Comment by Pil0tXia** [comment]
@kyooosukedn Sorry for the delay. May you please resolve conflicts? 😊

@Alonexc May you please help merge this PR?

**Comment by kyooosukedn** [comment]
@Alonexc @Pil0tXia kindly review this, thanks :)

**Review by Pil0tXia — APPROVED** [review]
Re-approval. cc @Alonexc

**Review by pchengma — APPROVED** [review]

**Comment by Pil0tXia** [comment]
Closing to re-trigger CI.

**Pull request merged**